jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js

```diff
@@ -23,12 +23,117 @@ YUI.add('deployment-credential', function() {
   juju.components.DeploymentCredential = React.createClass({
     propTypes: {
       acl: React.PropTypes.object.isRequired,
+      addTemplate: React.PropTypes.func.isRequired,
       cloud: React.PropTypes.string,
-      clouds: React.PropTypes.object.isRequired
+      clouds: React.PropTypes.object.isRequired,
+      credential: React.PropTypes.string,
+      listRegions: React.PropTypes.func.isRequired,
+      listTemplates: React.PropTypes.func.isRequired,
+      region: React.PropTypes.string,
+      setCredential: React.PropTypes.func.isRequired,
+      setRegion: React.PropTypes.func.isRequired,
+      setTemplate: React.PropTypes.func.isRequired,
+      template: React.PropTypes.string,
+      users: React.PropTypes.object.isRequired,
+      validateForm: React.PropTypes.func.isRequired
     },
 
+    credentialXHR: null,
+    regionsXHR: null,
```

Comment:
These two will need to move into the `getInitialState` method as properties defined here are shared across all instances of this component. While here it's not really a problem, it's a good practice to get into. It really belongs in an initializer but this `getInitialState` is the closest we get without using es6 classes.